components/post_view/post/index.js

```diff
@@ -81,6 +81,7 @@ function makeMapStateToProps() {
         return {
             post,
             createAriaLabel: createAriaLabelForPost(state, post),
+            currentChannel: channel,
```

Comment:
Instead of passing the entire channel object which may cause unnecessary re-render, you may pass what's needed only, could be:
```suggestion
            isDMChannel: Boolean(channel && channel.type === Constants.DM_CHANNEL),
```